pkg/standalone-utils/contracts/CowSwapFeeBurner.sol

```diff
@@ -0,0 +1,312 @@
+        return _getOrder(sellToken);
+    }
+
+    /// @inheritdoc ICowSwapFeeBurner
+    function getOrderStatus(IERC20 sellToken) external view returns (OrderStatus status) {
+        (status, ) = _getOrderStatusAndBalance(sellToken);
+    }
+
+    /// @inheritdoc ICowSwapFeeBurner
+    function retryOrder(IERC20 sellToken, uint256 minTargetTokenAmount, uint256 deadline) external authenticate {
+        (OrderStatus status, uint256 amount) = _getOrderStatusAndBalance(sellToken);
+
+        _checkDeadline(deadline);
+        _checkMinTargetTokenAmount(minTargetTokenAmount);
+
+        if (status != OrderStatus.Failed) {
+            revert OrderHasUnexpectedStatus(status);
+        }
+
+        _orders[sellToken].buyAmount = minTargetTokenAmount;
+        _orders[sellToken].validTo = uint32(deadline);
+
+        _createCowOrder(sellToken);
+
+        emit OrderRetried(sellToken, amount, minTargetTokenAmount, deadline);
+    }
+
+    /// @inheritdoc ICowSwapFeeBurner
+    function revertOrder(IERC20 sellToken, address receiver) external authenticate {
+        (OrderStatus status, uint256 amount) = _getOrderStatusAndBalance(sellToken);
+
+        if (status != OrderStatus.Failed) {
+            revert OrderHasUnexpectedStatus(status);
+        }
+
+        delete _orders[sellToken];
+
+        SafeERC20.safeTransfer(sellToken, receiver, amount);
+
+        emit OrderReverted(sellToken, receiver, amount);
+    }
+
+    function emergencyRevertOrder(IERC20 sellToken, address receiver) external authenticate {
+        delete _orders[sellToken];
+
+        uint256 amount = sellToken.balanceOf(address(this));
+        SafeERC20.safeTransfer(sellToken, receiver, amount);
+
+        emit OrderReverted(sellToken, receiver, amount);
+    }
+
+    /***************************************************************************
+                            IProtocolFeeBurner
+    ***************************************************************************/
+
+    /// @inheritdoc IProtocolFeeBurner
+    function burn(
+        address pool,
+        IERC20 feeToken,
+        uint256 feeTokenAmount,
+        IERC20 targetToken,
+        uint256 minTargetTokenAmount,
+        address recipient,
+        uint256 deadline
+    ) external authenticate {
+        if (targetToken == feeToken) {
+            revert InvalidOrderParameters("Fee token and target token are the same");
+        } else if (feeTokenAmount == 0) {
+            revert InvalidOrderParameters("Fee token amount is zero");
+        }
+
+        _checkDeadline(deadline);
+        _checkMinTargetTokenAmount(minTargetTokenAmount);
+
+        (OrderStatus status, ) = _getOrderStatusAndBalance(feeToken);
+        if (status != OrderStatus.NotExist && status != OrderStatus.Filled) {
+            revert OrderHasUnexpectedStatus(status);
+        }
+
+        SafeERC20.safeTransferFrom(feeToken, msg.sender, address(this), feeTokenAmount);
```

Comment:
```suggestion
```
This has been changed to transfer the tokens from the sweeper before calling burn, so that you don't need to pull them. (Otherwise we have a "hanging approval" issue.)